src/app/shared/auth-form/auth-form.component.ts

```diff
@@ -0,0 +1,59 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import {
+  AbstractControl,
+  FormControl,
+  FormGroup,
+  Validators,
+} from '@angular/forms';
+
+@Component({
+  selector: 'app-auth-form',
+  templateUrl: 'auth-form.component.html',
+})
+export class AuthFormComponent implements OnInit {
+  @Input() title?: string;
+  @Input() description?: string;
+  @Output() authHandler = new EventEmitter<{
+    email: string;
+    password: string;
+  }>();
+  signupForm: FormGroup = new FormGroup({});
+
+  ngOnInit() {
+    this.signupForm = new FormGroup({
+      email: new FormControl('', [
+        Validators.required,
+        this.emailValidator.bind(this),
+      ]),
+      password: new FormControl('', [
+        Validators.required,
+        this.passwordValidator.bind(this),
+      ]),
+    });
+  }
+
+  emailValidator(control: AbstractControl): { [s: string]: boolean } | null {
+    if (!control.value.includes('@')) {
```

Comment:
Seems like you can use just this validator intead of custom
https://angular.io/api/forms/Validators#email